zephyr/drivers/can/can_sccan.c

```diff
@@ -185,15 +231,91 @@ static int sc_can_disable(const struct sc_can_cfg *config)
 	return 0;
 }
 
+static uint32_t sc_can_get_idr(uint32_t id, bool extended, bool rtr)
+{
+	uint32_t idr;
+
+	if (extended) {
+		idr = (SCCAN_TXID_EX1(id) | SCCAN_TXSRTR(1) | SCCAN_TXIDE(extended) |
+		       SCCAN_TXID_EX2(id) | SCCAN_TXERTR(rtr));
+	} else {
+		idr = (SCCAN_TXID1(id) | SCCAN_TXSRTR(rtr) | SCCAN_TXIDE(extended));
+	}
+
+	return idr;
+}
+
+static void sc_can_tx_done(const struct device *dev, int status)
+{
+	struct sc_can_data *data = dev->data;
+	can_tx_callback_t callback;
+
+	callback = data->tx_cb[data->tx_tail];
+	if (callback != NULL) {
+		callback(dev, status, data->tx_cb_arg[data->tx_tail]);
+		data->tx_cb[data->tx_tail] = NULL;
+
+		data->tx_tail++;
+		if (data->tx_tail == SCCAN_TX_FIFO_DEPTH) {
+			data->tx_tail = 0;
+		}
```

Comment:
I'd do:
`data->tx_tail >= SCCAN_TX_FIFO_DEPTH`, or
`data->tx_tail = data->tx_tail++ % SCCAN_TX_FIFO_DEPTH` (modulo), or
`data->tx_tail = data->tx_tail++ & DEPTH_MASK` (if `SCCAN_TX_FIFO_DEPTH` is guaranteed to be a power of 2).

Reply:
will change to `data->tx_tail >= SCCAN_TX_FIFO_DEPTH`